Simplified/NYPLBookCellDelegate.m

```diff
@@ -175,70 +175,79 @@ - (void)openAudiobook:(NYPLBook *)book {
     dict[@"id"] = book.identifier;
   }
   
-  id<Audiobook> const audiobook = [AudiobookFactory audiobook: dict ?: json];
-
-  if (!audiobook) {
-    [self presentUnsupportedItemError];
-    return;
-  }
+  [AudioBookVendorsHelper updateVendorKeyWithBook:json completion:^(NSError * _Nullable error) {
+    [NSOperationQueue.mainQueue addOperationWithBlock:^{
+      id<Audiobook> const audiobook = [AudiobookFactory audiobook: dict ?: json];
+
+      if (!audiobook) {
+        if (error) {
+          [self presentDRMKeyError:error];
+        } else {
+          [self presentUnsupportedItemError];
+        }
+        return;
```

Comment:
do we need to log an error event to Crashlytics for both cases here? Not sure how hard it is to diagnose these errors (I know you're logging with `Log` but those are buried inside individual events whenever they are reported)

Reply:
`presentUnsupportedItemError` doesn't log to Crashalytics. Do you think we need to log all unsupported items to Crashalytics? Maybe we can create a separate ticket for that kind of issues, it is called for unsupported pdfs and any unknown book object.